packages/pluggableWidgets/barcode-scanner-web/src/hooks/useReader.ts

```diff
@@ -0,0 +1,76 @@
+import { useEffect, useRef, RefObject } from "react";
+import { BarcodeFormat, BrowserMultiFormatReader, DecodeHintType } from "@zxing/library/cjs";
+import { useEventCallback } from "@mendix/piw-utils-internal";
+
+const hints = new Map();
+// RSS_Expanded is not production ready yet.
+const exclusions = new Set([BarcodeFormat.RSS_EXPANDED]);
+// `BarcodeFormat` is a TypeScript enum. Calling `Object.values` on it returns an array of string and ints, we only want the latter.
+const formats = Object.values(BarcodeFormat)
+    .filter((format): format is BarcodeFormat => typeof format !== "string")
+    .filter(format => !exclusions.has(format));
+hints.set(DecodeHintType.POSSIBLE_FORMATS, formats);
+
+const mediaStreamConstraints: MediaStreamConstraints = {
+    audio: false,
+    video: {
+        facingMode: "environment",
+        width: { min: 1280, ideal: 1920, max: 2560 },
+        height: { min: 720, ideal: 1080, max: 1440 }
+    }
+};
+
+type UseReaderHook = (args: {
+    onSuccess?: (data: string) => void;
+    onError?: (e: Error) => void;
+}) => RefObject<HTMLVideoElement>;
+
+export const useReader: UseReaderHook = args => {
```

Comment:
`args` look clean, but I would really like to see the list of arguments
`export const useReader: UseReaderHook = ({onSuccess, onError})`

Reply:
I did this on purpose as on line 30 and 31 I use same names.

Reply:
I see now, then do as you like!